merkledag/coding.go

```diff
@@ -13,9 +13,9 @@ import (
 // for now, we use a PBNode intermediate thing.
 // because native go objects are nice.
 
-// Unmarshal decodes raw data into a *Node instance.
+// unmarshal decodes raw data into a *Node instance.
 // The conversion uses an intermediate PBNode.
-func (n *Node) Unmarshal(encoded []byte) error {
+func (n *Node) unmarshal(encoded []byte) error {
```

Comment:
I don't really remember why, but I try in this PR to reduce the services a `merkledag.Node` provides so it is easier to change it to IPLD.

Here, the problem is that when decoding a byte stream to an IPLD Node, it means a slightly different thing than decoding a merkledag Node from a byte stream.

- merkledag Node will intepret the byte stream as protocol buffers and decore it
- IPLD takes a byte stream, but it is a multicodec. That is a header information followed by protocol buffers or CBOR.
- IPLD can also take directly a byte stream without the multicodec header but you have to specify the encoding separately.

We can't just replace one function with the other. I thought it would be wise to keep this function internal to avoid using it elsewhere when it might be difficult to change. Currently its name is `Unmarshall` but in the future we could perhaps rename it to `UnmarshallProtoBuf` and have another function named `Unmarshall` that takes a multicodec byte stream.

Perhaps I'm over thinking it.